pkg/agent/multicluster/mc_route_controller_test.go

```diff
@@ -290,3 +426,177 @@ func TestMCRouteControllerAsRegularNode(t *testing.T) {
+	defer c.queue.ShutDown()
+	mockInterface.EXPECT().DeleteRouteForLink(gomock.Any(), gomock.Any()).Times(1)
+
+	testCases := []struct {
+		name        string
+		ciImport    *mcv1alpha1.ClusterInfoImport
+		expectError error
+	}{
+		{
+			name: "remove peer successfully",
+			ciImport: &mcv1alpha1.ClusterInfoImport{
+				Spec: mcv1alpha1.ClusterInfo{
+					ServiceCIDR: "10.100.0.0/16",
+				},
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		assert.Equal(t, tt.expectError, c.removeWireGuardRouteAndPeer(tt.ciImport))
+	}
+}
+
+func TestEnqueueGateway(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockInterface := routemock.NewMockInterface(ctrl)
+	networkConfig := &config.NetworkConfig{}
+	c := newMCDefaultRouteController(t,
+		&config.NodeConfig{
+			Name: "node-4",
+			PodIPv4CIDR: &net.IPNet{
+				IP: net.ParseIP("10.10.0.0/16"),
+			},
+		},
+		networkConfig,
+		agent.WireGuardConfig{},
+		mockInterface,
+		"wireGuard",
+	)
+	c.wireGuardClient = &wireguardmock.MockWireGuardClient{}
+	defer c.queue.ShutDown()
+
+	testCases := []struct {
+		name      string
+		obj       interface{}
+		isDeleted bool
+		expectNum int
+	}{
+		{
+			name:      "gateway is deleted, enqueue successfully",
+			obj:       &mcv1alpha1.Gateway{},
+			isDeleted: true,
+			expectNum: 1,
+		},
+		{
+			name: "gateway not deleted, enqueue successfully",
+			obj: &mcv1alpha1.Gateway{
+				InternalIP: "10.10.1.1",
+				GatewayIP:  "10.10.1.1",
+			},
+			isDeleted: true,
+			expectNum: 1,
+		},
+		{
+			name:      "gateway not deleted, enqueue failed",
+			obj:       &mcv1alpha1.Gateway{},
+			isDeleted: false,
+			expectNum: 0,
+		},
+		{
+			name:      "unexpect object",
+			obj:       map[string]string{},
+			isDeleted: false,
+			expectNum: 0,
+		},
+
+		{
+			name: "invalid delete state",
+			obj: cache.DeletedFinalStateUnknown{
+				Obj: map[string]string{},
+			},
+			isDeleted: false,
+			expectNum: 0,
+		},
+	}
+
+	for _, tt := range testCases {
+		c.queue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "gatewayroute")
+		c.enqueueGateway(tt.obj, tt.isDeleted)
+		assert.Equal(t, tt.expectNum, c.queue.Len())
+	}
+}
+
+func TestEnqueueClusterInfoImport(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockInterface := routemock.NewMockInterface(ctrl)
+	networkConfig := &config.NetworkConfig{}
+	c := newMCDefaultRouteController(t,
+		&config.NodeConfig{
+			Name: "node-4",
+			PodIPv4CIDR: &net.IPNet{
+				IP: net.ParseIP("10.10.0.0/16"),
+			},
+		},
+		networkConfig,
+		agent.WireGuardConfig{},
+		mockInterface,
+		"wireGuard",
+	)
+	c.wireGuardClient = &wireguardmock.MockWireGuardClient{}
+	defer c.queue.ShutDown()
+
+	testCases := []struct {
+		name      string
+		obj       interface{}
+		isDeleted bool
+		expectNum int
+	}{
+		{
+			name:      "ClusterInfoImport without GatewayInfo",
+			obj:       &mcv1alpha1.ClusterInfoImport{},
+			isDeleted: false,
```

Comment:
I didn't see any case cover `isDeleted: true`, if it doesn't impact the result, you may remove this field.